src/profile/v100/address/non_fungible_global_id.rs

```diff
@@ -105,6 +108,25 @@ impl From<NonFungibleGlobalId> for RETNonFungibleGlobalId {
     }
 }
 
+impl From<NonFungibleGlobalId> for ScryptoNonFungibleGlobalId {
+    fn from(value: NonFungibleGlobalId) -> Self {
+        Self::new(
+            value.resource_address.into(),
+            value.non_fungible_local_id.into(),
+        )
+    }
+}
+
+impl From<(ScryptoNonFungibleGlobalId, NetworkID)> for NonFungibleGlobalId {
```

Comment:
This is a pattern you will see ALOT in this PR, we must use:
```rust
From<(ScryptoNonFungibleGlobalId, NetworkID)>
```

instead of just:
```rust
From<ScryptoNonFungibleGlobalId>
```

Since Radix Engine (inside Scrypto repo) does NOT contain the NetworkID. 

And why do we need FROM Scrypto in the first place? Well this PR implements ExecutionSummary and use RET's exeuction summary, which contains Scrypto addresses.... so we must go FROM RET -> Sargon. And thus we need to pass a long the NetworkID **all over the place**.